### PR TITLE
Update external dependencies

### PR DESCRIPTION
**Pull request by pgrawehr**
Updates a set of external dependencies

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/iot/pull/2128)

### CODE REVIEW

File: eng/StyleCop.Analyzers.ruleset [hunk 1]
```diff
@@ -136,7 +136,7 @@
     <Rule Id="SA1515" Action="None" /> <!-- Single-line comment should be preceded by blank line -->
     <Rule Id="SA1516" Action="None" /> <!-- Elements should be separated by blank line -->
     <Rule Id="SA1517" Action="Error" />
-    <Rule Id="SA1518" Action="Error" />
+    <Rule Id="SA1518" Action="None" />
```
Comment (krwq) [inline eng/StyleCop.Analyzers.ruleset:139]: can you add a comment telling what this is? (only for ones your changing)

File: eng/Versions.external.props [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-    <PackageReference Include="UnitsNet" Version="5.1.0" />
+    <PackageReference Include="UnitsNet" Version="5.31.0" />
   </ItemGroup>
 
   <!-- Automatically include these assemblies in all test projects -->
   <ItemGroup Condition="$(MSBuildProjectName.EndsWith('Tests'))">
-     <PackageReference Include="Moq" Version="4.16.1" />
-     <PackageReference Include="xunit" Version="2.4.1" />
-     <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+     <PackageReference Include="Moq" Version="4.18.4" />
```
Comment (krwq) [inline eng/Versions.external.props:9]: last pause with this one for now

**Review by Ellerbach — APPROVED** [review]
Looks good to me!

File: eng/StyleCop.Analyzers.ruleset [hunk 3]
```diff
@@ -136,7 +136,7 @@
     <Rule Id="SA1515" Action="None" /> <!-- Single-line comment should be preceded by blank line -->
     <Rule Id="SA1516" Action="None" /> <!-- Elements should be separated by blank line -->
     <Rule Id="SA1517" Action="Error" />
-    <Rule Id="SA1518" Action="Error" />
```
Comment (joperezr) [inline eng/StyleCop.Analyzers.ruleset:139]: I'm fine with this change, but is there a specific reason why we decided to change it in this PR?

File: src/System.Device.Gpio.Tests/GpioPinTests.cs [hunk 4]
```diff
@@ -37,7 +37,7 @@ public void TestOpenPin()
             // Act
             GpioPin pin = ctrl.OpenPin(PinNumber, PinMode.Input);
             // Assert
-            Assert.Equal(pin.PinNumber, PinNumber);
```
Comment (joperezr) [inline src/System.Device.Gpio.Tests/GpioPinTests.cs:40]: Great catch 🙂

**Review by joperezr — COMMENTED** [review]
Changes looks good to me. Let's undo the change in NuGet.config so that we can validate which packages we need to push to our feed.

File: eng/StyleCop.Analyzers.ruleset (hunk 3 above)
Comment (pgrawehr) [reply eng/StyleCop.Analyzers.ruleset:139]: For some reason, the error started to occur for many files. Looks like the test didn't work previously and was now fixed with the update.

**Comment by pgrawehr** [comment]
@joperezr Yes, it's failing now (as expected). Please upload the missing pieces.

**Comment by joperezr** [comment]
I was finally able to push the package to the botnet-public feed so we should be good to go there. I can't re-trigger CI since there is a merge conflict at the moment, @pgrawehr can you take a look at that so that we can requeue this?

**Pull request merged**